tests/transforms/test_split_non_commuting.py

```diff
@@ -953,3 +955,123 @@ def test_trainable_hamiltonian_tensorflow(self, grouping_strategy):
         actual = tape.jacobian(cost, params)
 
         assert qml.math.allclose(actual, [-0.5, np.cos(np.pi / 4)])
+
+
+# Single hamiltonian expval measurement to test shot distribution
+ham = qml.Hamiltonian(
+    coeffs=[10, 0.1, 20, 100, 0.2],
+    observables=[
+        qml.X(0) @ qml.Y(1),
+        qml.Z(0) @ qml.Z(2),
+        qml.Y(1),
+        qml.X(1) @ qml.X(2),
+        qml.Z(0) @ qml.Z(1) @ qml.Z(2),
+    ],
+)
+
+
+class TestShotDistribution:
+    """
+    Test shot distribution for the `split_non_commuting` transform.
+    At the moment, this feature is available only for the single hamiltonian
+    measurement case and with "default" or "qwc" grouping_strategy.
+    """
+
+    @pytest.mark.parametrize("grouping_strategy", ["default", "qwc", "wires", None])
+    def test_none_shot_distribution(self, grouping_strategy):
+        """Test standard behaviour with no shot distribution strategy."""
+
+        total_shots = 1000
+        initial_tape = qml.tape.QuantumScript(measurements=[qml.expval(ham)], shots=total_shots)
+
+        tapes, _ = split_non_commuting(
+            initial_tape,
+            grouping_strategy=grouping_strategy,
+            shot_distribution=None,
+        )
+
+        assert sum(tape.shots.total_shots for tape in tapes) == total_shots * len(tapes)
+
+        for tape in tapes:
+            assert tape.shots.total_shots == total_shots
+
+    @pytest.mark.parametrize("shot_distribution", [0, 1.2, [], {}])
+    def test_type_error_shot_distribution(self, shot_distribution):
+        """Test an error is raised for shot_distribution incorrect type."""
+
+        initial_tape = qml.tape.QuantumScript(measurements=[qml.expval(ham)], shots=10)
+
+        with pytest.raises(
+            TypeError, match="`shot_distribution` must be a callable or str or None,"
+        ):
+            _ = split_non_commuting(initial_tape, shot_distribution=shot_distribution)
+
+    @pytest.mark.parametrize("grouping_strategy", ["default", "qwc"])
+    @pytest.mark.parametrize(
+        ["shot_distribution", "expected_shots"],
+        [
+            ("uniform", (334, 333, 333)),
+            ("weighted", (231, 2, 767)),
+            ("weighted_random", (221, 3, 776)),  # requires fixed seed=42
```

Comment:
I have the same concern here. Mind run a numeric simulation to collect some rough distribution, and finally make the assertion check a range with acceptance rate ~95%?